### ISSUE TITLE
Bug: Special eslint indent case not working seemingly

### ISSUE THREAD

**Issue by Droidking18** [eslint/eslint#15686]
### Environment

Node version: v14.15.0
npm version: v6.14.8
Operating System: darwin 21.3.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
"indent": ["error", 2],
```
</details>

```js
import
  React, {
  Component
} from 'react'
```

### What did you expect to happen?

This seemingly should be valid according to the rule 

### What actually happened?

Complains with `Expected indentation of 0 spaces but found 2.`.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @Droidking18, thanks for the issue!

I can reproduce this in our [online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwiaW5kZW50XCI6IFtcImVycm9yXCIsIDJdICovXG5cbmltcG9ydFxuICBSZWFjdCwge1xuICBDb21wb25lbnRcbn0gZnJvbSAncmVhY3QnXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19). The error is on the `React` line, and the expected indentation is:

```js
import
React, {
  Component
} from 'react'
```

In all test cases with default imports they're on the same line as the `import` keyword, so it looks like this case wasn't predicted.

It might make sense to apply [ImportDeclaration](https://eslint.org/docs/rules/indent#importdeclaration) option to default imports as well.

```js
/* eslint "indent": ["error", 2] */  // default "ImportDeclaration": 1

import
  React, {
  Component
} from 'react'
```

```js
/* eslint "indent": ["error", 2, { "ImportDeclaration": 2 }] */

import
    React, {
    Component
} from 'react'
```

It's getting complicated if `"ImportDeclaration"` is set to `"first"`, because it aligns named imports with the first named import. I think the rule should just ignore default imports in that case.

Alternatively, we could update the rule to always ignore `ImportDefaultSpecifier` nodes to allow users to indent them as they prefer, or keep the rule as it is now and recommend adding `"ImportDefaultSpecifier"` to `ignoredNodes` if the current behavior isn't desirable (here's a [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwiaW5kZW50XCI6IFtcImVycm9yXCIsIDIsIHsgaWdub3JlZE5vZGVzOiBbXCJJbXBvcnREZWZhdWx0U3BlY2lmaWVyXCJdIH1dICovXG5cbmltcG9ydFxuICBSZWFjdCwge1xuICBDb21wb25lbnRcbn0gZnJvbSAncmVhY3QnXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19) with `ignoredNodes: ["ImportDefaultSpecifier"]`).

I'd appreciate opinions from other team members on what seems the best course of action.

1. Update the rule to require a different indentation for `ImportDefaultSpecifier` nodes, based on the existing `ImportDeclaration` option.
2. Update rule to require a different indentation for `ImportDefaultSpecifier` nodes, based on something else (maybe a new option).
3. Update the rule to ignore `ImportDefaultSpecifier` nodes.
4. Keep the current behavior and recommend `ignoredNodes: ["ImportDefaultSpecifier"]`

The same should probably apply to `ImportNamespaceSpecifier` nodes.

**Comment by Gautam-Arora24** [comment]
I would like to work on this issue :)

**Comment by Gautam-Arora24** [comment]
Is this issue ready to be implemented?

**Comment by nzakas** [comment]
I think option 4 makes the most sense to me. The fewer changes to the indent rule, the better, IMHO

**Comment by mdjermanovic** [comment]
I agree with @nzakas. The `ignoredNodes` option can be used when user disagrees with the indentation that the rule enforces for a particular syntactic pattern, and this seems like a good case for this option.

The solution is to add `ImportDefaultSpecifier` selector to `ignoredNodes` in your configuration for the `indent` rule:

```
"indent": ["error", 2, { ignoredNodes: ["ImportDefaultSpecifier"] }]
```
